src/data/equiv/algebra.lean

```diff
@@ -328,3 +327,73 @@ instance symm.is_ring_hom {e : α ≃r β} : is_ring_hom e.to_equiv.symm := hom
   e.symm.to_equiv x = e.to_equiv.symm x := rfl
 
 end ring_equiv
+
+-- We first define the type of automorphisms on groups, additive monoids,
+-- additive groups and rings using `mul_equiv`, `add_equiv` and `ring_equiv`.
+-- In each case this type is also a group.
+
+namespace group
+
+def aut (γ : Type) [group γ] := mul_equiv γ γ
+
+@[extensionality] lemma ext_mul_equiv (α β : Type) [group α] [group β] (f : mul_equiv α β)
+(g : mul_equiv α β) (h : f.to_fun = g.to_fun) : f = g :=
+by { cases f, cases g, congr, apply equiv.eq_of_to_fun_eq h }
+
+instance aut_group (γ : Type) [group γ] : group (aut γ) :=
+{ mul := λ g h, mul_equiv.trans h g,
```

Comment:
Maybe add a comment explaining why `h` and `g` are "backwards" here.

Reply:
fixed